### ISSUE TITLE
typescript compiler error on adding dragHandleProps

### ISSUE THREAD

**Issue by shamimfahad** [atlassian/react-beautiful-dnd#2116]
Everything was working fine till a couple of days ago. I thought it might be caused by version upgrade but seems I'm still using v11.0.5. 

This is the error I'm getting 
```TypeScript
Type '{ children: string; className: string; style: { whiteSpace: "nowrap"; textOverflow: string; width: string; }; } | { children: string; onMouseDown: (event: MouseEvent<any, MouseEvent>) => void; ... 10 more ...; style: { ...; }; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>'.
  Type '{ children: string; onMouseDown: (event: MouseEvent<any, MouseEvent>) => void; onKeyDown: (event: KeyboardEvent<any>) => void; onTouchStart: (event: TouchEvent<any>) => void; onTouchMove: (event: TouchEvent<...>) => void; ... 7 more ...; style: { ...; }; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>'.
    Type '{ children: string; onMouseDown: (event: MouseEvent<any, MouseEvent>) => void; onKeyDown: (event: KeyboardEvent<any>) => void; onTouchStart: (event: TouchEvent<any>) => void; onTouchMove: (event: TouchEvent<...>) => void; ... 7 more ...; style: { ...; }; }' is not assignable to type 'HTMLAttributes<HTMLHeadingElement>'.
      Types of property 'onDragStart' are incompatible.
        Type '(event: DragEvent<any>) => void' is not assignable to type '(event: DragEvent<HTMLHeadingElement>) => void'.
          Types of parameters 'event' and 'event' are incompatible.
            Type 'DragEvent<HTMLHeadingElement>' is missing the following properties from type 'DragEvent<any>': detail, view  TS2322
```

I guess this error is only generated on my machine. 

I'm using react v16.9.0.

And react-beautiful-dnd v11.0.5.

I use chrome.

I'm facing problem with the following block and a similar one

```TypeScript
<>
    <Draggable
      draggableId={props.column._id}
      key={props.column._id}
      index={props.index}
    >
      {(provided) => (
        <div
          className="dib h-100 w-23-l w-50-m"
          style={{ minWidth: '270px' }}
          {...provided.draggableProps}
          ref={provided.innerRef}
        >
          <div className="flex flex-column h-100 pr3 w-100">
            <div className="w-100">
              <div className="flex pb1 w-100">
                <h3
                  className="pl1 overflow-hidden ma0 fw6 flex-grow-1 ws-normal ws-break pv1"
                  style={{
                    whiteSpace: 'nowrap',
                    textOverflow: 'ellipsis',
                    width: '220px',
                  }}
                  {...provided.dragHandleProps}
                >
                  {props.column.name}
                </h3>
              </div>
            </div>
    
            <div
              className="h-100 mb1 overflow-y-auto"
              style={{
                background: '#F6F6F6',
                borderRadius: '0 0 5px 5px',
              }}
            >
              <div className="flex-grow-1 flex flex-column h-100">
                <Droppable
                  droppableId={props.column._id!}
                  type="task"
                >
                  {(provided, snapshot) => {
                    const allowedProps = { ref: provided.innerRef };
                    return (
                      <div
                        className={`pa2 flex-grow-1 flex
                  flex-column overflow-auto 
                  ${
                    snapshot.isDraggingOver
                      ? 'bg-lightest-blue'
                      : ''
                  }`}
                        style={{
                          transition: 'background-color 0.2s ease',
                          minHeight: '100px',
                          whiteSpace: 'normal',
                        }}
                        {...provided.droppableProps}
                        {...allowedProps}
                      >
                        <div className="flex-grow-1">
                          {props.column.tasks!.map(
                            (task, index) => (
                              <TaskItem
                                key={task._id!}
                                task={task}
                                index={index}
                              />
                            ),
                          )}
                        </div>
                        {provided.placeholder}
                      </div>
                    );
                  }}
                </Droppable>
              </div>
            </div>
          </div>
        </div>
      )}
    </Draggable>
</>
```

**Comment by danieldelcore** [comment]
Hi @shamimfahad 👋, the issue seems to be related to this code:

```jsx
            <h3
                 ...
                  {...provided.dragHandleProps}
                >
```

Seems to not like the heading element types with the event for some reason. 

Can you try Upgrading to `@types/react-beautiful-dnd` versrion 12?

**Comment by danieldelcore** [comment]
For reference typescript definitions are available here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-beautiful-dnd/v12/index.d.ts

**Comment by shamimfahad** [comment]
hi @danieldelcore, I have updated `@types/react-beautiful-dnd` to version 12.1.0 
![image](https://user-images.githubusercontent.com/48179840/111601215-560af880-87fc-11eb-878a-2032f3b8d1ff.png)
 but the error is still there

**Comment by amisrs** [comment]
I'm also seeing the same error with 
```
    "@types/react-beautiful-dnd": "^13.0.0",
    "react-beautiful-dnd": "^13.0.0"
```

Specifying the provided's type as `any` works though.

e.g. this compiles and works normally
```
{(provided: any)} => (
  <div 
    ref={provided.innerRef} 
    {...provided.dragHandleProps} 
    {...provided.draggableProps} />    
)
```
Specifying the type as DraggableProvided results in the same error.

**Comment by henriqueleite42** [comment]
I find out that If you are using [NextJS](https://nextjs.org/), the `next-env.d.ts` in the `include` of `tsconfig.json` file causes this bug.

If you remove it from the `include`, TS won't complain anymore, but I don't know the full consequences of this.

Before (this causes the bug):
![image](https://user-images.githubusercontent.com/50992814/112060684-43fec200-8b3c-11eb-9763-8eccd7aae2bc.png)

After (bug don't happens anymore):
![image](https://user-images.githubusercontent.com/50992814/112060740-5416a180-8b3c-11eb-839d-5be48484133a.png)

`package.json` dependencies, for aditinal info:
![image](https://user-images.githubusercontent.com/50992814/112061136-d1daad00-8b3c-11eb-8d62-7f7d06789a00.png)

**Comment by danieldelcore** [comment]
Hey @shamimfahad, can you give @henriqueleite42's solution a try and let me know if it works for you?

**Comment by danieldelcore** [comment]
Also, i've setup a small repro here: https://codesandbox.io/s/holy-currying-4cvgj?file=/src/App.tsx 

Seems to be working well in isolation, but i might be missing something. Have a look for me 😄

**Comment by shamimfahad** [comment]
@danieldelcore I think there was a problem in my yarn.lock file in that branch, somehow it got corrupted. Everyone else working on that repo did not have that error. Neither could they generate it. So I cloned my repo into a new folder and if I don't ever checkout that specific branch, everything's working fine. Then I moved the codes from there to a new branch and deleted that branch. 

thanks a lot for your help :)
